file_changed_command_runner/callback_server.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import threading
+
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+
+log = logging.getLogger(__name__)
+
+
+class CallbackHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200, "OK")
+        self.end_headers()
+        self.close_connection = True  # pylint: disable=attribute-defined-outside-init
+
+        self.server: "CallbackServer"
+        self.server.process_callback(self.path.lstrip("/"))
+
+
+class CallbackServer(HTTPServer):
+    """Listens on localhost for a request with a specific token"""
+
+    port_range = range(50000, 51000)
```

Comment:
What's the advantage of this vs letting the OS choose an unused port?

Reply:
none what so ever. didn't realize i could pass in 0 to have an unused port selected for me. much easier.